src/Main.ps1

```diff
@@ -1011,6 +1011,8 @@ function Invoke-Pester {
                 $pluginConfiguration["Coverage"] = $CodeCoverage
             }
 
+            if ($SCRIPT:additionalPlugins) {$plugins += $SCRIPT:additionalPlugins}
```

Comment:
```suggestion
            if ($script:additionalPlugins) {$plugins += $script:additionalPlugins}
```

Reply:
> This would not allow you to add a plugin in the middle of our built-in plugins, or remove other plugins, hope that is okay for your usecase. You could use a list of scriptblocks, that would take list of plugins and give you another list, but that gets complicated fast.
> 
> Please add a note that this is a quick hack for test adapter, and that no-one should depend on it. And if they do then they should let us know in this issue: #2021 to help drive the future plugins public api.


How about this approach then that basically lets you attach a callback to manipulate the plugins array, then you can do whatever you need to inject one or more plugins wherever you need. Obviously this isn't "safe" but it's pretty much nothing you can't do manipulating the module anyways :)

```powershell
  # Enable the ability to manipulate the plugin array to inject custom plugins via a callback
  # Example Usage:
  # & (gmo pester) {
  #     $SCRIPT:PluginHandler = {
  #     param($Plugins)
      
  #     $InjectionPoint = $Plugins.IndexOf($Plugins | Where-Object Name -eq 'RSpecObjectDecoratorPlugin')
  #     $Plugins.Insert($InjectionPoint, New-PluginObject -Name 'MyCustomPlugin')
  # }

  if ($SCRIPT:PluginHandler) {
      . $SCRIPT:PluginHandler -Plugins $Plugins
  }
```

Reply:
Yeah something like what I had in mind. Have you tried it thought? For me it was pretty complicated to produce a plugin that would still bind correctly to the correct script and possibly have some external state. But maybe I was just going too deep into it.

Reply:
@nohwnd I haven't tried it via this specific method, but here's my current shim implementation that is "backwards compatible"

https://github.com/pester/vscode-adapter/blob/cf4a100401eae0582fa6f0614cf1a3ca6822c918/Scripts/PesterInterface.ps1#L289-L348